### PR TITLE
Array binary search

### PR DESCRIPTION
**Pull request by gmireles210**
- [ ] Top-level README "Table of Contents" is updated
 - [ ] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
     - [ ] "Happy Path" - Expected outcome
     - [ ] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [ ] README for this challenge is complete
     - [ ] Summary, Description, Approach & Efficiency, Solution
     - [ ] Link to code
     - [ ] Picture of whiteboard

### CODE REVIEW

**Comment by mnfmnfm** [comment]
Please merge this PR so your code is on your master branch!

File: code401challenges/src/main/java/code401challenges/BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package code401challenges;
+
+import java.util.Arrays;
+
+public class BinarySearch {
+    public static int BinaryMachine(int[] startArray, int pinpointNum) {
```
Comment (mnfmnfm) [inline code401challenges/src/main/java/code401challenges/BinarySearch.java]: Nitpick: We name all methods with lowerCamelCase, not UpperCamelCase; we save UpperCamelCase for class names.

File: code401challenges/src/test/java/code401challenges/BinarySearchTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package code401challenges;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+public class BinarySearchTest {
+    @Test public void testBinaryMachine() {
+        int[] arr = {1, 2, 3, 4, 5, 6};
+        int key = 1;
+        int actualOutput = BinarySearch.BinaryMachine(arr, key);
+        System.out.println("actualOutput = " + actualOutput);
```
Comment (mnfmnfm) [inline code401challenges/src/test/java/code401challenges/BinarySearchTest.java]: For this to be an actual test, it needs to contain an assertion, not just a System.out.println. So in this case, that might look like:
`assertEquals("should find 1 at index 0", 0, actualOutput);`
